File: app/src/main/java/com/shveed/cookmegood/activity/StartActivity.java
package com.shveed.cookmegood.activity;

import android.content.Intent;
import androidx.annotation.NonNull;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;

import com.shveed.cookmegood.entity.User;
import com.shveed.cookmegood.interfaces.FragmentChangeListener;
import com.shveed.cookmegood.fragment.CabinetFragment;
import com.shveed.cookmegood.fragment.CartFragment;
import com.shveed.cookmegood.fragment.FavouritesFragment;
import com.shveed.cookmegood.fragment.MainFragment;
import com.shveed.cookmegood.fragment.SuggestFragment;
import com.shveed.wallpapperparser.R;

import java.util.ArrayList;
import java.util.List;

public class StartActivity extends FragmentActivity implements FragmentChangeListener {

    private User currentUser;

    private Fragment selectedFragment;

    private List<Fragment> fragments = new ArrayList<>();

    private BottomNavigationView.OnNavigationItemSelectedListener navListener =
            new BottomNavigationView.OnNavigationItemSelectedListener() {
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            selectedFragment = new MainFragment();
            switch (item.getItemId()) {
                case R.id.navigation_suggest:
                    selectedFragment = fragments.get(0);
                    break;
                case R.id.navigation_favourites:
                    selectedFragment = fragments.get(1);
                    break;
                case R.id.navigation_recipe:
                    selectedFragment = fragments.get(2);
                    break;
                case R.id.navigation_cart:
                    selectedFragment = fragments.get(3);
                    break;
                case R.id.navigation_profile:
                    selectedFragment = fragments.get(4);
                    break;
            }
            getSupportFragmentManager().beginTransaction().replace(R.id.fragmentStartFrameLayout, selectedFragment)
            .commit();
            return true;
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start);

        fragments.add(new SuggestFragment());
        fragments.add(new FavouritesFragment());
        fragments.add(new MainFragment());
        fragments.add(new CartFragment());
        fragments.add(new CabinetFragment());

        getSupportFragmentManager().beginTransaction().replace(R.id.fragmentStartFrameLayout, new MainFragment())
                .commit();

        BottomNavigationView navView = findViewById(R.id.nav_view_start);
        navView.setOnNavigationItemSelectedListener(navListener);

        currentUser = (User)getIntent().getSerializableExtra("userObject");
    }

    @Override
    public void replaceFragment(Fragment fragment) {
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.fragmentStartFrameLayout, fragment, fragment.toString());
        fragmentTransaction.addToBackStack(fragment.toString());
        fragmentTransaction.commit();
    }

    public void toSearch(View view){
        Intent intent = new Intent(this, SearchActivity.class);
        startActivity(intent);
    }

    public void toCategories(View view){
        getSupportFragmentManager().beginTransaction().replace(R.id.fragmentStartFrameLayout, new MainFragment())
                .commit();
    }
}

File: app/src/main/java/com/shveed/cookmegood/activity/StartActivity.kt
package com.shveed.cookmegood.activity

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.shveed.cookmegood.R
import com.shveed.cookmegood.fragment.*
import com.shveed.cookmegood.listener.FragmentChangeListener
import kotlinx.android.synthetic.main.activity_start.*


class StartActivity: FragmentActivity(), FragmentChangeListener  {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_start)
        nav_view_start.setOnNavigationItemSelectedListener(mOnNavigationItemSelected)
        supportFragmentManager.beginTransaction().replace(R.id.fragmentStartFrameLayout, MainFragment())
                .commit()
     //   val currentUser = intent.getSerializableExtra("userObject") as User
    }
    private val mOnNavigationItemSelected = BottomNavigationView.OnNavigationItemSelectedListener {menuItem -> when
        (menuItem.itemId){
        R.id.navigation_suggest ->{
            val fragment = SuggestFragment()
            replaceFragment(fragment)
            Log.d("tag","sjd")
            return@OnNavigationItemSelectedListener true
        }
        R.id.navigation_favourites ->{
            val fragment = FavouritesFragment()
            replaceFragment(fragment)
            return@OnNavigationItemSelectedListener true
        }
        R.id.navigation_cart ->{
            val fragment = CartFragment()
            replaceFragment(fragment)
            return@OnNavigationItemSelectedListener true
        }
        R.id.navigation_profile ->{
            val fragment = CabinetFragment()
            replaceFragment(fragment)
            return@OnNavigationItemSelectedListener true
        }
        R.id.navigation_recipe ->{
            val fragment = MainFragment()
            replaceFragment(fragment)
            return@OnNavigationItemSelectedListener true
        }
    };false
    }

    override fun replaceFragment(fragment: Fragment) {
        val fragmentManager: FragmentManager = supportFragmentManager
        val fragmentTransaction: FragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.fragmentStartFrameLayout, fragment, fragment.toString())
        fragmentTransaction.addToBackStack(fragment.toString())
        fragmentTransaction.commit()
    }

    fun toSearch(view: View?) {
        val intent = Intent(this, SearchActivity::class.java)
        startActivity(intent)
    }

    fun toCategories(view: View?) {
        supportFragmentManager.beginTransaction().replace(R.id.fragmentStartFrameLayout, MainFragment())
                .commit()
    }

}